src/Paket.Core/RemoteDownload.fs

```diff
@@ -132,10 +147,41 @@ let rec DirectoryCopy(sourceDirName, destDirName, copySubDirs) =
         for subdir in dirs do
             DirectoryCopy(subdir.FullName, Path.Combine(destDirName, subdir.Name), copySubDirs)
 
-/// Gets a single file from github.
+/// Retrieves RemoteFiles
 let downloadRemoteFiles(remoteFile:ResolvedSourceFile,destination) = async {
     match remoteFile.Origin, remoteFile.Name with
+    | SingleSourceFileOrigin.GitLink cloneUrl, _ -> 
+        let cloneUrl = cloneUrl.TrimEnd('/')
+        
+        let repoCacheFolder = Path.Combine(Constants.GitRepoFolder,remoteFile.Project)
+
+        // clone/fetch to cache
+        if Directory.Exists repoCacheFolder then
+            verbosefn "Fetching %s to %s" cloneUrl repoCacheFolder 
```

Comment:
Please note, there will be no trace message (non verbose) when the cache folder is already present.
With the other comment in mind, about forcing the download, the cache should be ignored in this case, right?